flytekit/common/mixins/registerable.py

```diff
@@ -73,3 +80,55 @@ def instantiated_in(self):
         :rtype: Optional[Text]
         """
         return self._instantiated_in
+
+    @property
+    def has_valid_name(self):
+        """
+        :rtype: bool
+        """
+        return self._platform_valid_name is not None and self._platform_valid_name != ""
+
+    @property
+    def platform_valid_name(self):
+        """
+        :rtype: Text
+        """
+        return self._platform_valid_name
+
+    def auto_assign_name(self):
+        """
+        This function is a bit of trickster Python code that goes hand in hand with the _InstanceTracker metaclass
+        defined above. Thanks @matthewphsmith for this bit of ingenuity.
```

Comment:
lol don't blame me for this. git blame will tell the truth :p

Reply:
😈